components/eam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -0,0 +1,1697 @@
+    ptend%q(:ncol,:pver,ixnumice)  = ( max(0._rtype,numice(:ncol,:pver) ) - state%q(:ncol,:pver,ixnumice)  )/dtime
+    ptend%q(:ncol,:pver,ixcldrim)  = ( max(0._rtype,qm(:ncol,:pver)  ) - state%q(:ncol,:pver,ixcldrim)  )/dtime
+    ptend%q(:ncol,:pver,ixrimvol)  = ( max(0._rtype,rimvol(:ncol,:pver) ) - state%q(:ncol,:pver,ixrimvol)  )/dtime
+
+    ! Update t_prev and qv_prev to be used by evap_precip
+    t_prev(:ncol,:pver) = temp(:ncol,:pver)
+    qv_prev(:ncol,:pver) = qv(:ncol,:pver)
+
+    call t_stopf('micro_p3_tend_loop')
+    call t_startf('micro_p3_tend_finish')
+   ! Following MG interface as a template:
+
+    ! Net micro_p3 condensation rate
+    qme(:ncol,top_lev:pver) = cmeliq(:ncol,top_lev:pver) + qv2qi_depos_tend(:ncol,top_lev:pver)  ! qv2qi_depos_tend is output from p3 micro
+    ! Add cmeliq to  vap_liq_exchange
+    vap_liq_exchange(:ncol,top_lev:pver) = vap_liq_exchange(:ncol,top_lev:pver) + cmeliq(:ncol,top_lev:pver) 
+
+!====================== Export variables/Conservation START ======================!
+     !For precip, accumulate only total precip in prec_pcw and snow_pcw variables.
+    ! Other precip output variables are set to 0
+    ! Do not subscript by ncol here, because in physpkg we divide the whole
+    ! array and need to avoid an FPE due to uninitialized data.
+    prec_pcw = precip_liq_surf + precip_ice_surf
+    prec_sed = 0._rtype
+    prec_str = prec_pcw + prec_sed
+
+    snow_pcw = precip_ice_surf
+    snow_sed = 0._rtype
+    snow_str = snow_pcw + snow_sed
+!====================== Export variables/Conservation END ======================!
+
+!====================== Radiation Specific Outputs START ======================!
+
+   ! Calculate rho for size distribution
+   ! parameter calculations and average it if needed
+   
+   rho(:ncol,top_lev:) = &
+      state%pmid(:ncol,top_lev:) / (rair*temp(:ncol,top_lev:))
+   ! ------------------------------------------------------------ !
+   ! Compute in cloud ice and liquid mixing ratios                !
+   ! Note that 'iclwp, iciwp' are used for radiation computation. !
+   ! ------------------------------------------------------------ !
+      
+   icinc = 0._rtype
+   icwnc = 0._rtype
+      
+   do k = top_lev, pver
+      do icol = 1, ncol
+         ! Limits for in-cloud mixing ratios consistent with P3 microphysics
+         ! in-cloud mixing ratio maximum limit of 0.005 kg/kg
+         icimrst(icol,k)   = min( state%q(icol,k,ixcldice) / max(mincld,cld_frac_i(icol,k)),0.005_rtype )
+         icwmrst(icol,k)   = min( state%q(icol,k,ixcldliq) / max(mincld,cld_frac_l(icol,k)),0.005_rtype )
+         icinc(icol,k)     = state%q(icol,k,ixnumice) / max(mincld,cld_frac_i(icol,k)) * &
+              state%pmid(icol,k) / (287.15_rtype*state%t(icol,k))
+         icwnc(icol,k)     = state%q(icol,k,ixnumliq) / max(mincld,cld_frac_l(icol,k)) * &
+              state%pmid(icol,k) / (287.15_rtype*state%t(icol,k))
+      end do                    
+   end do
```

Comment:
@crterai @Jiwengithub I remembered that Jiwen talked about the issue in diagnosing Nc in P3 on Last Friday's meeting. Is it this part? Are we going make another mini PR for that? Thank you.

Reply:
@mingxuanwupnnl, I don't recall the issue, but @Jiwengithub might remember.

Reply:
@mingxuanwupnnl Yes, this is the issue. Currently, the output incloud properties do not include MP tendencies updated. We suggested to change the code to: 
         icimrst(icol,k)   = min( ice(icol,k) / max(mincld,cld_frac_i(icol,k)),0.005_rtype )
         icwmrst(icol,k)   = min( cldliq(icol,k) / max(mincld,cld_frac_l(icol,k)),0.005_rtype )
         icinc(icol,k)     = numice(icol,k) / max(mincld,cld_frac_i(icol,k)) * rho(icol,k) !&
         icwnc(icol,k)     = numliq(icol,k) / max(mincld,cld_frac_l(icol,k)) * rho(icol,k) !&